src/agent/application.cpp

```diff
@@ -40,23 +40,29 @@
 #include "agent/application.hpp"
 #include "common/code_utils.hpp"
 #include "common/mainloop_manager.hpp"
+#include "utils/infra_link_selector.hpp"
 
 namespace otbr {
 
 bool                 Application::sShouldTerminate = false;
 const struct timeval Application::kPollTimeout     = {10, 0};
 
 Application::Application(const std::string &              aInterfaceName,
-                         const std::string &              aBackboneInterfaceName,
+                         const std::vector<const char *> &aBackboneInterfaceNames,
                          const std::vector<const char *> &aRadioUrls)
     : mInterfaceName(aInterfaceName)
-    , mBackboneInterfaceName(aBackboneInterfaceName)
-    , mNcp(mInterfaceName.c_str(), aRadioUrls, mBackboneInterfaceName.c_str(), /* aDryRun */ false)
+#if __linux__
```

Comment:
We are using `__linux__` because the current code compiles and works on Linux only.
We can remove `#if __linux__` when other platforms are supported. 

I think currently we don't encourage adding configurations as compile options.